### PR TITLE
added two broken test cases to clarify error handling

### PR DESCRIPTION
**Pull request by ghost**

### CODE REVIEW

File: test/epipe_tests.erl [hunk 1]
```diff
@@ -59,9 +59,27 @@ pipe_test_() ->
...
+            Fun2 = fun(_Val) -> erlang:error("Bad CRC")  end,
+
+            ?assertEqual(
+                {error, step2, "Can't process data", 1},
+                epipe:run([{step1, Fun1}, {step2, Fun2}, {step3, Fun1}], 0)
+            )
+        end},
+        {"Function throws an error", fun() ->
+            Fun1 = fun(Val) -> {ok, Val + 1} end,
+            Fun2 = fun(_Val) -> throw("Terrible CRC")  end,
```
Comment (oltarasenko) [inline test/epipe_tests.erl:78]: This does not follow the spec of the run function https://github.com/oltarasenko/epipe/blob/master/src/epipe.erl#L10-L19. See also http://erlang.org/doc/man/erlang.html#exit-1

File: test/epipe_tests.erl [hunk 2]
```diff
@@ -59,9 +59,27 @@ pipe_test_() ->
...
             Fun2 = fun(_Val) -> {error, "Can't process data"} end,
 
+            ?assertEqual(
+                {error, step2, "Can't process data", 1},
+                epipe:run([{step1, Fun1}, {step2, Fun2}, {step3, Fun1}], 0)
+            )
+        end},
+        {"Function raises an error", fun() ->
+            Fun1 = fun(Val) -> {ok, Val + 1} end,
+            Fun2 = fun(_Val) -> erlang:error("Bad CRC")  end,
```
Comment (oltarasenko) [inline test/epipe_tests.erl:69]: Again, this does not follow the spec of run function.

**Comment by oltarasenko** [comment]
In general, I would not want to do error handling inside the pipe runner itself. This task should be deligated to steps instead.

**Comment by ghost** [comment]
Understood. Maybe cover this point in the blog post so it's clear. B

**Pull request closed without merging**